jobbergate-api/jobbergate_api/apps/clusters/routers.py

```diff
@@ -62,3 +62,26 @@ async def get_cluster_status(
     logger.debug("Getting list of cluster statuses")
     query = select(ClusterStatus).order_by(ClusterStatus.client_id)
     return await paginate(secure_session.session, query)
+
+
+@router.get(
+    "/status/{client_id}",
+    description="Endpoint to get the status of a specific cluster.",
+    response_model=ClusterStatusView,
+)
+async def get_cluster_status_by_client_id(
+    client_id: str,
+    secure_session: SecureSession = Depends(
+        secure_session(Permissions.ADMIN, Permissions.CLUSTERS_READ, commit=False)
+    ),
+):
+    """
+    Get the status of a specific cluster.
+    """
+    logger.debug("Getting cluster status for client_id={}", client_id)
+    query = select(ClusterStatus).where(ClusterStatus.client_id == client_id)
+    result = await secure_session.session.execute(query)
+    instance = result.scalar_one_or_none()
+    if instance is None:
+        return FastAPIResponse(status_code=status.HTTP_404_NOT_FOUND)
+    return ClusterStatusView.model_validate(instance)
```

Comment:
I don't believe it's necessary to do `model_validate()` here as FastAPI should do this automatically.

Reply:
I didn't understand actually. Do you mean `ClusterStatusView(instance)` should be enough or `return instance` would work?

The `instance` object has type `ClusterStatus`, which is a SQLAlchemy ORM model. I'm not sure if FastAPI would handle it accordingly. As well as, I don't think it's best practice in terms of coding to return a SQLAlchemy ORM model.

Reply:
FastAPI can handle the ORM model internally, other endpoints in the project are coded in this way as well, so I agree with Tucker for consistency

Reply:
I understand FastAPI can handle an ORM model internally, however, this pattern is a bit weird to me. It makes useless the concept of validators and serializers then, and if I use somewhere in the code specifically because of a serializer let's say, it would be inconsistent with other endpoints.

That said, the modification was made in [77da40b](https://github.com/omnivector-solutions/jobbergate/pull/591/commits/77da40b5209da556a9dc30fc58d3abd6e38ab6d4).